File: packages/create-llama/templates/components/vectordbs/typescript/pg/generate.mjs
/* eslint-disable turbo/no-undeclared-env-vars */
import * as dotenv from "dotenv";
import {
  PGVectorStore,
  SimpleDirectoryReader,
  VectorStoreIndex,
  storageContextFromDefaults,
} from "llamaindex";
import {
  PGVECTOR_SCHEMA,
  PGVECTOR_TABLE,
  STORAGE_DIR,
  checkRequiredEnvVars,
} from "./shared.mjs";

dotenv.config();

async function loadAndIndex() {
  // load objects from storage and convert them into LlamaIndex Document objects
  const documents = await new SimpleDirectoryReader().loadData({
    directoryPath: STORAGE_DIR,
  });

  // create postgres vector store
  const vectorStore = new PGVectorStore({
    connectionString: process.env.PG_CONNECTION_STRING,
    schemaName: PGVECTOR_SCHEMA,
    tableName: PGVECTOR_TABLE,
  });
  vectorStore.setCollection(STORAGE_DIR);
  vectorStore.clearCollection();

  // create index from all the Documents
  console.log("Start creating embeddings...");
  const storageContext = await storageContextFromDefaults({ vectorStore });
  await VectorStoreIndex.fromDocuments(documents, { storageContext });
  console.log(`Successfully created embeddings.`);
}

(async () => {
  checkRequiredEnvVars();
  await loadAndIndex();
  console.log("Finished generating storage.");
  process.exit(0);
})();

File: packages/create-llama/templates/components/vectordbs/python/pg/generate.py
from dotenv import load_dotenv

load_dotenv()

import logging
from llama_index.core.indices import VectorStoreIndex
from llama_index.core.storage import StorageContext

from app.engine.constants import DATA_DIR
from app.engine.loader import get_documents
from app.engine.settings import init_settings
from app.engine.utils import init_pg_vector_store_from_env

logging.basicConfig(level=logging.INFO)
logger = logging.getLogger()


def generate_datasource():
    logger.info("Creating new index")
    # load the documents and create the index
    documents = get_documents()
    store = init_pg_vector_store_from_env()
    storage_context = StorageContext.from_defaults(vector_store=store)
    VectorStoreIndex.from_documents(
        documents,
        storage_context=storage_context,
        show_progress=True,  # this will show you a progress bar as the embeddings are created
    )
    logger.info(
        f"Successfully created embeddings in the PG vector store, schema={store.schema_name} table={store.table_name}"
    )


if __name__ == "__main__":
    init_settings()
    generate_datasource()
